File: API/Models/Property.cs
using System.ComponentModel.DataAnnotations;

namespace Semsar_online.Models
{
    public class Property
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string SellerId { get; set; }
        [Required]
        public int NumberOfRooms { get; set; }
        [Required]
        public float Area { get; set; }
        
        [Required]
        public int Price { get; set; }

        [Required]
        public string Location { get; set; }

        [Required]
        public string Type { get; set; }
        
        public string? Description { get; set; }
        [Required]
        public DateTime ListingDate { get; set; } = DateTime.Now;

        [Required]
        public int DownPayment {  get; set; }
        [Required]
        public string Status { get; set; }

        public bool deleted { get; set; } = false;
        public string[] Images {get;set;}
        public DateTime? DeletedAt { get; set; }

        public void Delete()
        {
            deleted = true;
            DeletedAt = DateTime.Now;
        }
        public Company? Company { get; set; }
        public ICollection<Review>? Reviews { get; set; }
        public ICollection<Appointment>? Appointments { get; set; }
    }
}

File: frontend/src/app/Interfaces/property.ts
import { Company } from "./company";

export interface Property{
  id?: 1,
  numberOfRooms?: 4,
  area?: 2000,
  price?: 120000,
  Location?: 'New York',
  type?: 'House',
  description?: 'This is a beautiful house in New York',
  listingDate?: '2021-01-01',
  downPayment?: 20000,
  status?: 'For Sale',
  images?: string[],
  company?: Company
}
